qontract_api/tests/external/ldap/test_router.py

```
@@ -0,0 +1,105 @@
+    resolved from app.state. This fixture sets both mocks so the dependencies
+    resolve without a live cache or Vault connection.
+    """
+    from qontract_api.main import app
+
+    app.state.cache = Mock()
+    app.state.secret_manager = Mock()
+
+    yield TestClient(app, raise_server_exceptions=False)
+
+    if hasattr(app.state, "cache"):
+        delattr(app.state, "cache")
```

Ruff (0.15.9)

[warning] Do not call `delattr` with a constant attribute value. It is not any safer than normal property deletion.

Replace `delattr` with `del` statement

(B043)